frontend/src/metabase/visualizations/components/ChartSettings/BaseChartSettings/hooks.ts

```diff
@@ -61,22 +65,30 @@ export const useChartSettingsSections = ({
     [currentSection, sectionNames, sections],
   );
 
-  const visibleWidgets = sections[chartSettingCurrentSection] || [];
+  const visibleWidgets = useMemo(
+    () => sections[chartSettingCurrentSection] || [],
+    [chartSettingCurrentSection, sections],
+  );
```

Comment:
I guess the reason this needs to be memoized is because of `|| []` creating new empty arrays every render?

Reply:
yeah that's the idea anyway. Looking at it again, I think it might be an over-optimization but I don't think it hurts us here